src/User/RegistrationForm.jsx

```diff
@@ -85,6 +87,8 @@ const RegistrationForm = () => {
             label="Username: "
             type="text"
             name="username"
+            pattern="^[a-zA-Z0-9]*$"
+            title="Must not contain any spaces."
```

Comment:
This also doesn't allow special characters, so you might want to put that in the error message as well :)